utils/generator/transformers/enums_producer.py

```diff
@@ -0,0 +1,118 @@
+"""
+Enums transformation
+"""
+
+import logging
+import textwrap
+from collections import namedtuple, OrderedDict
+
+from model.enum import Enum
+from model.enum_element import EnumElement
+from transformers.common_producer import InterfaceProducerCommon
+
+
+class EnumsProducer(InterfaceProducerCommon):
+    """
+    Enums transformation
+    """
+
+    def __init__(self, paths):
+        super(EnumsProducer, self).__init__(
+            container_name='elements',
+            enums_package=None,
+            structs_package=None,
+            package_name=paths.enums_package)
+        self.logger = logging.getLogger('EnumsProducer')
+        self._params = namedtuple('params', 'origin name internal description since value deprecated')
+
+    @staticmethod
+    def converted(name):
+        if name[0].isdigit():
+            name = '_' + name
+        if '-' in name:
+            name = name.replace('-', '_')
+        return name
+
+    def transform(self, item: Enum) -> dict:
+        """
+        Override
+        :param item: particular element from initial Model
+        :return: dictionary to be applied to jinja2 template
+        """
+        imports = set()
+        params = OrderedDict()
+        if any(map(lambda l: l.name != self.converted(l.name) and getattr(l, 'value', None) is not None,
+                   getattr(item, self.container_name).values())):
```

Comment:
I am just wondering about what enums have the `complex`  kind? I don't see any of them do. However, if I made a case like that in the `MOBILE_API.xml`. Like this one, for example, the generator will not work well in that case:

```
<enum name="PredefinedWindows" since="6.0">
    <element name="0DEFAULT_WINDOW" value="0">
        <description>The default window is a main window pre-created on behalf of the app.</description>
    </element>
    <element name="1PRIMARY_WIDGET" value="1">
        <description>The primary widget of the app.</description>
    </element>
</enum>
```
Maybe I missing something.

Reply:
the idea of that type was to cover cases like that: 
```bash
sdl_java_suite/base/src/main/java/com/smartdevicelink/protocol/enums/FunctionID.java
```
```java
SYNC_P_DATA(65537, "SyncPData"),
```
but as we agreed to skip such cases, I agree that `complex` kind is useless.
Deleting it.